code/game/objects/structures/crates_lockers/closets/bodybag.dm

```diff
@@ -147,3 +148,163 @@
 		max_weight_of_contents = A_is_item.w_class
 	B.w_class = max_weight_of_contents
 	usr.put_in_hands(B)
+
+/// Environmental bags
+
+/obj/structure/closet/body_bag/environmental
+	name = "environmental protection bag"
+	desc = "An insulated, reinforced bag designed to protect against exoplanetary storms and other environmental factors."
+	icon = 'icons/obj/bodybag.dmi'
+	icon_state = "envirobag"
+	mob_storage_capacity = 1
+	pressure_protection = 0.8
+	thermal_insulation = 0.5
+	foldedbag_path = /obj/item/body_bag/environmental/
+	weather_protection = list("acid", "ash", "rad", "snow", "void") // Does not protect against lava or the The Floor Is Lava spell.
+
+/obj/structure/closet/body_bag/environmental/nanotrasen
+	name = "elite environmental protection bag"
+	desc = "Used for VIP extraction from hazardous environments and as shelter by elite Nanotrasen operatives, this highly reinforced bag is nearly completely isolated from environmental factors when closed."
+	icon = 'icons/obj/bodybag.dmi'
+	icon_state = "ntenvirobag"
+	pressure_protection = 1
+	thermal_insulation = 1
+	foldedbag_path = /obj/item/body_bag/environmental/nanotrasen/
+	weather_protection = list("all")
+
+/// Securable enviro. bags
+
+/obj/structure/closet/body_bag/environmental/prisoner
+	name = "prisoner transport bag"
+	desc = "Intended for transport of prisoners through hazardous environments, this environmental protection bag comes with straps to keep an occupant secure."
+	icon = 'icons/obj/bodybag.dmi'
+	icon_state = "prisonerenvirobag"
+	foldedbag_path = /obj/item/body_bag/environmental/prisoner/
+	breakout_time = 3000 // Five minutes, because it's probably about as hard to get out of this as it is to get out of a straightjacket.
+	var/sinch_time = 100
+	var/sinched = FALSE
+
+/obj/structure/closet/body_bag/environmental/prisoner/update_icon()
+	. = ..()
+	if(sinched)
+		icon_state = initial(icon_state) + "_sinched"
+	else
+		icon_state = initial(icon_state)
+
+/obj/structure/closet/body_bag/environmental/prisoner/can_open(mob/living/user, force = FALSE)
+	if(force)
+		return TRUE
+	if(sinched)
+		to_chat(user, span_danger("The buckles on [src] are sinched down, preventing it from opening."))
+		return FALSE
+	. = ..()
+
+/obj/structure/closet/body_bag/environmental/prisoner/open(mob/living/user, force = FALSE)
+	if(!can_open(user, force))
+		return
+	if(opened)
+		return
+	sinched = FALSE
+	playsound(loc, open_sound, open_sound_volume, TRUE, -3)
+	opened = TRUE
+	if(!dense_when_open)
+		set_density(FALSE)
+	dump_contents()
+	update_appearance()
+	after_open(user, force)
+	return TRUE
+
+/obj/structure/closet/body_bag/environmental/prisoner/container_resist_act(mob/living/user)
+	/// copy-pasted with changes because flavor text as well as some other misc stuff
+	if(opened)
+		return
+	if(ismovable(loc))
+		user.changeNext_move(CLICK_CD_BREAKOUT)
+		user.last_special = world.time + CLICK_CD_BREAKOUT
+		var/atom/movable/AM = loc
```

Comment:
better var name please

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names